src/main/kotlin/io/github/anki/anki/controller/ControllerExceptionHandler.kt

```diff
@@ -1,24 +1,47 @@
 package io.github.anki.anki.controller
 
+import io.github.anki.anki.controller.exceptions.CardDoesNotExistException
+import io.github.anki.anki.controller.exceptions.DeckDoesNotExistException
+import io.github.anki.anki.controller.exceptions.DoesNotExist
 import org.slf4j.LoggerFactory
 import org.springframework.http.HttpStatus
 import org.springframework.http.ResponseEntity
+import org.springframework.validation.FieldError
+import org.springframework.web.HttpRequestMethodNotSupportedException
+import org.springframework.web.bind.MethodArgumentNotValidException
 import org.springframework.web.bind.annotation.ExceptionHandler
-import org.springframework.web.bind.annotation.ResponseStatus
 import org.springframework.web.bind.annotation.RestControllerAdvice
+import java.util.LinkedHashMap
 
 @RestControllerAdvice
 class ControllerExceptionHandler {
+    @ExceptionHandler(HttpRequestMethodNotSupportedException::class)
+    fun methodNotSupportedHandler(): ResponseEntity<String> =
+        ResponseEntity.status(HttpStatus.NOT_FOUND).body("Not Found")
+
+    @ExceptionHandler(MethodArgumentNotValidException::class)
+    fun handleValidationExceptions(ex: MethodArgumentNotValidException): ResponseEntity<Map<String, String>> =
+        ResponseEntity(getValidationException(ex), HttpStatus.BAD_REQUEST)
+
+    @ExceptionHandler(DeckDoesNotExistException::class, CardDoesNotExistException::class)
+    fun doesNotExistHandler(ex: DoesNotExist): ResponseEntity<String> =
+        ResponseEntity.status(HttpStatus.BAD_REQUEST).body(ex.message)
 
     @ExceptionHandler(Exception::class)
-    @ResponseStatus(value = HttpStatus.INTERNAL_SERVER_ERROR)
     fun globalExceptionHandler(
         ex: Exception,
     ): ResponseEntity<String> {
         LOG.error("Handling global exception", ex)
         return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("Internal Server Error")
     }
 
+    fun getValidationException(
```

Comment:
1) Will be better to create mapper
```kotlin
fun MethodArgumentNotValidException.toMap(): Map<String, String> {
        return bindingResult.fieldErrors
            .associate { it.field to (it.defaultMessage ?: "Validation error") }
    }
```
2) Move to `controller/mapper` package
3) Why have u used linked hash map? (it has reduced performance compared to map)

Reply:
i like toMap, i will change to it